source/renderer/dialog/note-network.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        NoteNetwork class
+ * CVM-Role:        View
+ * Maintainer:      Julien Mirval
+ * License:         GNU GPL v3
+ *
+ * Description:     This dialog lets you visualize all notes as a network.
+ *
+ * END HEADER
+ */
+
+const d3 = require('d3')
+
+const ZettlrDialog = require('./zettlr-dialog.js')
+
+class NoteNetwork extends ZettlrDialog {
+  constructor () {
+    super()
+    this._dialog = 'note-network'
+    this.graph = []
+  }
+
+  preInit (data) {
+    var tmpLinks = []
+    var tmpNodes = []
+
+    // Building link list
+    for (const a of Object.entries(data)) {
```

Comment:
Maybe consider renaming `a` to `idNodePair` and `i` to `targetNodeId` (if I got these correctly, that is).